rosbridge_library/src/rosbridge_library/internal/services.py

```diff
@@ -122,7 +127,19 @@ def call_service(node_handle, service, args=None):
 
     client = node_handle.create_client(service_class, service)
 
-    result = client.call(inst)
+    future = client.call_async(inst)
+    if sleep_time == 0:
```

Comment:
Maybe flip these to cover negative values?

```
if sleep_time > 0:
  # Do sleeping work
else:
  ...
```

Unless rclpy will do it for you...